src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -125,6 +146,14 @@ public InstanceConfiguration(String namePrefix,
         this.description = description;
         this.startupScript = startupScript;
         this.preemptible = preemptible;
+
+        this.windows = windows;
+        this.windowsConfig = createWindowsConfig(windowsUsername, windowsPasswordCredentialsId, windowsPrivateKeyCredentialsId);
```

Comment:
You can simplify createWindowsConfig by putting the call inside an if block here instead of having the method check the instance variable, it better preserves the orthogonality of the method.

Reply:
In-fact, with the above change createWindowsConfig can be removed, just use WindowsConfiguration constructor directly.